components/prism-hoon.js

```diff
@@ -8,6 +8,7 @@ Prism.languages.hoon = {
 	'class-name': [
 		{
 			pattern: /@(?:[A-Za-z0-9-]*[A-Za-z0-9])?/,
+			greedy: true
```

Comment:
`greedy: true` is one way to fix it. However, `greedy: true` is more expensive performance-wise and there is an even simpler fix: Just swap the order of `function` and `class-name`.

Swapping them should have the same result without the performance hit.